internal/server/server.go

```diff
@@ -76,9 +76,8 @@ func (s *Server) Ping(ctx context.Context, sender *pb.Node) (*pb.Node, error) {
 	}, nil
 }
 
-func (s *Server) FindNode(ctx context.Context, kademliaID *pb.KademliaID) (*pb.Nodes, error) {
-	// TODO this needs to make sure that sender node and itself is not included in response
-	candidates := s.routingTable.FindClosestContacts((*kademliaid.KademliaID)(kademliaID.Value), env.BucketSize)
+func (s *Server) FindNode(ctx context.Context, request *pb.FindNodeRequest) (*pb.Nodes, error) {
+	candidates := s.routingTable.FindClosestContacts((*kademliaid.KademliaID)(request.Target.Value), (*kademliaid.KademliaID)(request.Sender.Value), env.BucketSize)
```

Comment:
1. This is hard to read.
2. It is not very intuitive, if we have our own in the contact list. Because we need to have smth. like a "blacklist" attribute, which is not nice.